### PR TITLE
[Tests-Only] Add interactions for the tests in sharing tests

### PR DESCRIPTION
**Pull request by jasson99**
## Description:
New interactions and the interactions previously added in the pactHelper related to authorized sharing are added in the sharing tests file.

## Related Issue:
- https://github.com/owncloud/owncloud-sdk/issues/500#issuecomment-701138139

### CODE REVIEW

File: tests/sharingTest.js [hunk 1]
```diff
@@ -168,6 +295,51 @@ fdescribe('Main: Currently testing file/folder sharing,', function () {
...
+          promises.push(provider.addInteraction({
+            uponReceiving: 'a request to update a public link share',
+            withRequest: {
+              method: 'PUT',
+              path: Pact.Matchers.regex({
+                matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/\\d+$',
+                generate: '/ocs/v1.php/apps/files_sharing/api/v1/shares/9'
+              }),
+              headers: validAuthHeaders,
+              body: 'password=test123'
```
Comment (kiranparajuli589) [inline tests/sharingTest.js:313]: use `testUserPassword` var here

**Comment by kiranparajuli589** [comment]
lgtm :+1:

File: tests/sharingTest.js [hunk 2]
```diff
@@ -8,15 +8,7 @@ fdescribe('Main: Currently testing file/folder sharing,', function () {
...
-  const { setGeneralInteractions } = require('./pactHelper.js')
-
-  beforeAll(function (done) {
-    Promise.all(setGeneralInteractions(provider)).then(done, done.fail)
-  })
-
-  afterAll(function (done) {
-    provider.removeInteractions().then(done, done.fail)
-  })
+  const { validAuthHeaders, ocsMeta, shareResponseOcsData, accessControlAllowHeaders, accessControlAllowMethods, setGeneralInteractions } = require('./pactHelper.js')
```
Comment (individual-it) [inline tests/sharingTest.js:11]: too long line, hard to review

File: tests/sharingTest.js [hunk 3]
```diff
@@ -168,6 +295,51 @@ fdescribe('Main: Currently testing file/folder sharing,', function () {
...
+                'Access-Control-Allow-Origin': origin
+              },
+              body: '<?xml version="1.0"?>\n' +
+                '<ocs>\n' +
+                ocsMeta('ok', '100') +
+                ' <data>\n' +
+                shareResponseOcsData(3, 9, 1, '/testFolder') +
+                '  <share_with>***redacted***</share_with>\n' +
+                '  <share_with_displayname>***redacted***</share_with_displayname>\n' +
+                '  <name>..ffentlicher Link</name>\n' +
```
Comment (individual-it) [inline tests/sharingTest.js:327]: why the dots?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/sharingTest.js (hunk 3 above)
Comment (jasson99) [reply tests/sharingTest.js:391]: This is exactly what the name is when run in core server. `Öffentlicher` is shown as `..ffentlicher`

Comment (individual-it) [reply tests/sharingTest.js:391]: also when you try to query it with curl?
My guess would be that wireshark did not display the `Ö` correctly. I can check tomorrow

Comment (jasson99) [reply tests/sharingTest.js]: fixed

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/sharingTest.js [hunk 4]
```diff
@@ -8,46 +8,75 @@ fdescribe('Main: Currently testing file/folder sharing,', function () {
...
+  const sharedFilesByLink = {
+    'test space and + and #.txt': 18,
+    'test.txt': 14,
+    '文件.txt': 19
+  }
+  const sharedFilesWithGroup = {
+    'test space and + and #.txt': 18,
+    'test.txt': 14,
+    '文件.txt': 19
+  }
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/sharingTest.js]: fixed

File: tests/sharingTest.js [hunk 5]
```diff
@@ -196,120 +222,271 @@ fdescribe('Main: Currently testing file/folder sharing,', function () {
...
+              generate: '/ocs/v1.php/apps/files_sharing/api/v1/shares'
+            }),
+            headers: validAuthHeaders,
+            body: 'shareType=0&shareWith=' + config.testUser + '&path=' + config.testFilesPath[i] + '&expireDate=' + config.expirationDate
+          },
+          willRespondWith: {
+            status: 200,
+            headers: {
+              'Content-Type': 'application/xml; charset=utf-8',
+              'Access-Control-Allow-Origin': origin
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/sharingTest.js]: created a new variable for `application/xml` now

**Review by swoichha — COMMENTED** [review]
LGTM :+1:

**Review by individual-it — APPROVED** [review]

**Pull request merged**